### DISCUSSION TITLE
ts-run, a lightweigth alternative to tsimp

### DISCUSSION THREAD

**Discussion by Septh** [avajs/ava#3303] [Show and tell]
In case any one is interested, I have written [ts-run](https://github.com/Septh/ts-run), a minimalist TypeScript script runner for Node that can be used in lieu of the recommended `tsimp`.

Among other features, `ts-run` is light (only 220 KB on disk) and fast and it works in both ESM and CommonJS environments.

It is not as feature-full as `tsimp` but I use it successfully with ava to run the tests for some other project of mine. If you're willing to give it a try, I'd be happy if you'd let me know what you think.

**Comment by PopGoesTheWza** [comment]
Many thanks @Septh !

I was stuck for a couple of days trying to switch from `ts-node` to `tsimp` or whatever would work.

I am sharing this [test repo](https://github.com/PopGoesTheWza/test-ts-xo-ava) to illustrate my working configuration.

**Comment by PopGoesTheWza** [comment]
@tommy-mitchell maybe this solution is worth mentioning in #3335 ?
